src/enzyme_ad/jax/Passes/LLVMToMemrefAccess.cpp

```diff
@@ -0,0 +1,159 @@
+#include "Passes.h"
+
+#include "mlir/Analysis/TopologicalSortUtils.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Dialect/Vector/IR/VectorOps.h"
+#include "mlir/IR/MLIRContext.h"
+#include "src/enzyme_ad/jax/Dialect/Ops.h"
+
+#define DEBUG_TYPE "llvm-to-memref-access"
+
+namespace mlir {
+namespace enzyme {
+#define GEN_PASS_DEF_LLVMTOMEMREFACCESSPASS
+#include "src/enzyme_ad/jax/Passes/Passes.h.inc"
+} // namespace enzyme
+} // namespace mlir
+
+using namespace mlir;
+
+using PtrVal = TypedValue<LLVM::LLVMPointerType>;
+
+namespace mlir {
+MemRefType recoverMemRefTypeFromKernelCall(enzymexla::KernelCallOp kernelCallOp,
+                                           unsigned argIndex) {
+  auto tensorType =
+      kernelCallOp.getInputs()[argIndex].getType().dyn_cast<RankedTensorType>();
+  if (!tensorType) {
+    return nullptr;
+  }
+  // Convert tensor type to memref type
+  return MemRefType::get(tensorType.getShape(), tensorType.getElementType());
+}
+
+} // namespace mlir
+
+struct LLVMToMemrefAccessPass
+    : public enzyme::impl::LLVMToMemrefAccessPassBase<LLVMToMemrefAccessPass> {
+  using LLVMToMemrefAccessPassBase::LLVMToMemrefAccessPassBase;
+
+  void runOnOperation() override {
+    Operation *op = getOperation();
+
+    SymbolTableCollection symbolTableCollection;
+    SymbolTable symbolTable = symbolTableCollection.getSymbolTable(op);
+
+    MLIRContext *context = &getContext();
+
+    // Find all enzymexla.kernel_call ops and recover memref types for callees
+    op->walk([&](enzymexla::KernelCallOp kernelCallOp) {
+      auto *symbolOp = symbolTableCollection.lookupNearestSymbolFrom(
+          kernelCallOp, kernelCallOp.getFnAttr());
```

Comment:
We should have things apply to both KernelCAll and JitCall ops

Reply:
Added a todo for this, couldn't finish this today, will try to land this first to unblock.
Will work on this tomorrow